### ISSUE TITLE
Nagios Core 4.4.6 with Raspbian 11 [Bullseye] on NRPE 4.0.3 Failing - Anyone else?

### ISSUE THREAD

**Issue by discarn8** [NagiosEnterprises/nagioscore#858]
I upgraded my Pi3 to Raspbian 11 (Bullseye), the other night and immediately, Nagios Core started throwing errors for most of my checks:

"_(No output on stdout) stderr: corrupted size vs. prev_size while consolidating_"

Is the v4.0.3 plugin package compatible with  Nagios Core 4.4.6?

Raspberry Pi3 info:
$ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye

$ apt list --installed | grep nagios
nagios-nrpe-plugin/stable,now 4.0.3-1 armhf [installed]
nagios-nrpe-server/stable,now 4.0.3-1 armhf [installed]

$ nrpe -V
NRPE - Nagios Remote Plugin Executor
Version: 4.0.3

----------------------------------------------------

Nagios Core Information [Server]:
$ sudo /usr/local/nagios/bin/nagios –help
Nagios Core 4.4.6

$ nrpe -V
NRPE - Nagios Remote Plugin Executor
Version: 3.2.1

$ apt list | grep -i installed | grep -i nagios
nagios-nrpe-plugin/oldstable,now 3.2.1-2 armhf [installed]
nagios-nrpe-server/oldstable,now 3.2.1-2 armhf [installed]

$ cat /etc/os-release 
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster

Anyone else having this problem?  Thanks in advance

**Comment by FoUStep** [comment]
I have the exact same files and also running the same OS. No issues here whatsoever.

**Comment by discarn8** [comment]
Interesting. [Thank you for the reply]

May I ask you to compare the nrpe.cfg on one of your clients, to mine, please?

$ cat nrpe.cfg | egrep -v '#|^$'
log_facility=daemon
debug=0
pid_file=/var/run/nagios/nrpe.pid
server_port=5666
nrpe_user=nagios
nrpe_group=nagios
allowed_hosts=127.0.0.1,192.168.0.50,::1
dont_blame_nrpe=0
allow_bash_command_substitution=0
command_timeout=60
connection_timeout=300
include=/etc/nagios/nrpe_local.cfg
include_dir=/etc/nagios/nrpe.d/
command[check_users]=/usr/lib/nagios/plugins/check_users -w 3 -c 4
command[check_load]=/usr/lib/nagios/plugins/check_load -r -w 1.25,1.10,1.00 -c 1.40,1.35,1.20
command[check_hda1]=/usr/lib/nagios/plugins/check_disk -w 20% -c 10% -p /
command[check_zombie_procs]=/usr/lib/nagios/plugins/check_procs -w 5 -c 10 -s Z
command[check_total_procs]=/usr/lib/nagios/plugins/check_procs -w 250 -c 300

**Comment by FoUStep** [comment]
Almost the same config except for the commands ofcourse (I customized those).

I don't use IPv6 (so no ::1 in allowed hosts, also turned it off in sysctl) and I use SSL so I have several SSL commands in there (key, privkey, ca file and loggin). Perhaps it's trying over IPv6?

**Comment by discarn8** [comment]
FoUtep - you're a genius - thank you.

You got me thinking about TLS - and, after doing a tcpdump on my client traffic, I realized that the server requests were coming in, requesting TLS - the client responses were less than elegant. This reminded me of a similar issue I had with one of my clients **way back** and I dug through my old client cfg files on the server and found one of them had a "-2" after the check command. This forces v2 packets only:

" -2, --v2-packets-only        Only use version 2 packets, not version 3/4" 

I'm not sure why this host required it or why it fixed it, but it did - and that is all I am concerned with, at the moment.

* One thing to note - the previous client I had to do this with, used the non-conventional naming for the ethernet port. The current client, that I am doing this with, also has a non-conventional ethernet naming. Possibly a mere coincidence, possibly the culprit. That is for another A.D.H.D. induced rabbit-hole, some other time.

Thanks very much for your help!
